operator/backrest/restore.go

```diff
@@ -349,6 +325,15 @@ func CreateRestoredDeployment(restclient *rest.RESTClient, cluster *crv1.Pgclust
 	// set up a map of the names of the tablespaces as well as the storage classes
 	tablespaceStorageTypeMap := operator.GetTablespaceStorageTypeMap(cluster.Spec.TablespaceMounts)
 
+	// combine supplemental groups from all volumes
+	var supplementalGroups []int64
+	{
+		supplementalGroups = append(supplementalGroups, dataVolume.SupplementalGroups...)
+		for _, v := range tablespaceVolumes {
+			supplementalGroups = append(supplementalGroups, v.SupplementalGroups...)
+		}
+	}
```

Comment:
Similarly, not sure how I feel about this grouping. Is there cited precedent for using it in this way (e.g. Effective go?)

Reply:
Nothing in Effective Go. I've seen it used to control scope, often in main() or other "wiring" code, where some intermediate values (e.g. parsed config) are needed only briefly.

Nothing like that here.